src/metabase/channel/email/messages.clj

```diff
@@ -269,14 +269,26 @@
     (email/send-message! message)))
 
 (defn generate-pulse-unsubscribe-hash
-  "Generates hash to allow for non-users to unsubscribe from pulses/subscriptions."
+  "Generates hash to allow for non-users to unsubscribe from pulses/subscriptions.
+
+  Deprecated: only used for dashboard subscriptions for now, should be migrated to `generate-notification-unsubscribe-hash`
+  once we migrate all the dashboard subscriptions to the new notification system."
   [pulse-id email]
   (codecs/bytes->hex
    (encryption/validate-and-hash-secret-key
     (json/encode {:salt     (public-settings/site-uuid-for-unsubscribing-url)
                   :email    email
                   :pulse-id pulse-id}))))
 
+(defn generate-notification-unsubscribe-hash
+  "Generates hash to allow for non-users to unsubscribe from notifications."
+  [notification-id email]
+  (codecs/bytes->hex
+   (encryption/validate-and-hash-secret-key
+    (json/encode {:salt            (public-settings/site-uuid-for-unsubscribing-url)
+                  :email           email
+                  :notification-id notification-id}))))
```

Comment:
I dislike that it's not bound by time, so any link generated will be used indefinitely. Not sure this is worth changing here, but it's certainly a security issue.